create/approve.py

```diff
@@ -70,43 +72,122 @@ def wait_for_task(celery, task):
 
         print('.', end='')
         sys.stdout.flush()
+
     print()
+
     if isinstance(task.result, Exception):
         raise task.result
     else:
         return task.result
 
 
-def main():
-    def_group_name = ''
-    def_callink_oid = ''
-    def_email = ''
-
-    parser = ArgumentParser(description='Create new OCF group accounts.')
-    parser.add_argument('oid', type=int, nargs='?', help='CalLink OID for the group.')
-    args = parser.parse_args()
+def get_group_information(group_oid):
+    if group_oid:
+        group = group_by_oid(group_oid)
 
-    if args.oid:
-        group = group_by_oid(args.oid)
         if not group:
-            print(red('No group with OID {}').format(args.oid))
-            return
+            print(red('No group with OID {}').format(group_oid))
+            sys.exit(1)
+
         if group['accounts']:
             print(yellow(
                 'Warning: there is an existing group account with OID {}: {}'.format(
-                    args.oid,
+                    group_oid,
                     ', '.join(group['accounts']),
                 ),
             ))
-            input('Press any key to continue...')
-        def_group_name = group['name']
-        def_callink_oid = args.oid
-        def_email = group['email']
+            input('Press enter to continue...')
+
+        return (group['name'], group_oid, group['email'])
+    else:
+        return ('', '', '')
```

Comment:
Yeah, I definitely agree on this, it would help to have all the printing things in one method. It was tough with this script though, because a lot of it is based around printing things, so I split it up this way.